tools/jolie2java/src/joliex/java/impl/JavaDocumentCreator.java

```
@@ -606,27 +595,38 @@ private void constructorCreate(StringBuilder stringBuilder, TypeDefinition type/
 
                     }
 
+                } else if (subType instanceof TypeChoiceDefinition) {
+                    //can't initialize variable with several possible types
```

We should throw an error here.